src/js/core/factories/GridRenderContainer.js

```diff
@@ -1,793 +1,808 @@
+        var self = this;
 
-    var minCols = self.minColumnsToRender();
+        var minRows = self.minRowsToRender();
+        var rowCache = self.visibleRowCache;
+        var maxRowIndex = rowCache.length - minRows;
+        var vertLen = self.getVerticalScrollLength();
 
-    var columnCache = self.visibleColumnCache;
-    var maxColumnIndex = columnCache.length - minCols;
+        //// Calculate the scroll percentage according to the scrollTop location, if no percentage was provided
+        if ((typeof(scrollPercentage) === 'undefined' || scrollPercentage === null) && scrollTop) {
+          scrollPercentage = scrollTop / vertLen ;
+        }
 
-    // Calculate the scroll percentage according to the scrollLeft location, if no percentage was provided
-    if ((typeof(scrollPercentage) === 'undefined' || scrollPercentage === null) && scrollLeft) {
-      scrollPercentage = scrollLeft / self.getHorizontalScrollLength();
-    }
+        var theoreticalRowIndex = 0;
+        var heightAtFirstRow = 0;
+        var currentHeight = 0;
+        var i = 0;
+
+        rowCache.forEach(function(r){
+          if ( scrollTop > currentHeight && scrollTop < currentHeight + r.height){
+            theoreticalRowIndex = i;
+            heightAtFirstRow = currentHeight;
+          }
+
+          currentHeight+= r.height;
+          i++;
+        });
+
+        var rowIndex = 0; // OLD = Math.ceil(Math.min(maxRowIndex, maxRowIndex * scrollPercentage));
+        rowIndex = Math.min(maxRowIndex,theoreticalRowIndex);
+
+        //TODO: do we need to fix here?
+        // Define a max row index that we can't scroll past
+        // if (rowIndex > maxRowIndex) {
+        //   rowIndex = maxRowIndex;
+        // }
+
+        var newRange = [];
+        if (rowCache.length > self.grid.options.virtualizationThreshold) {
+
+          if (!(typeof(scrollTop) === 'undefined' || scrollTop === null)) {
+            // Have we hit the threshold going down?
+            if ( !self.grid.suppressParentScrollDown && self.prevScrollTop < scrollTop && rowIndex < self.prevRowScrollIndex + self.grid.options.scrollThreshold && rowIndex < maxRowIndex) {
+              console.log('returning first bail');
+              return;
+            }
+            //Have we hit the threshold going up?
+            if ( !self.grid.suppressParentScrollUp && self.prevScrollTop > scrollTop && rowIndex > self.prevRowScrollIndex - self.grid.options.scrollThreshold && rowIndex < maxRowIndex) {
+              console.log('returning second bail');
+              return;
+            }
+          }
+
+          var rangeStart = {};
+          // var totalNumRowsToRender = Math.min(rowCache.length,rowIndex + minRows + self.grid.options.excessRows);
+
+          //TODO: @vance calculate and appropriate end... doesn't break things, just show more rows than necessary here.
+          var rangeEnd = rowCache.length;
```

Comment:
Showing more rows than necessary is considered a defect for some companies. Can you describe what you mean by that? Will the scrolling end, or will it just keep going since this is the infinite scroll library?